client/task-list/tasks/payments/index.js

```diff
@@ -216,28 +230,46 @@ class Payments extends Component {
 		} );
 	}
 
+	getSetupElement( method ) {
+		return cloneElement( method.container, {
+			query: this.props.query,
+			installStep: this.getInstallStep( method ),
+			markConfigured: this.markConfigured,
+			markConfigurationFinished: this.markConfigurationFinished.bind(
+				this,
+				method.key
+			),
+			hasCbdIndustry: method.hasCbdIndustry,
+			key: 'setup-' + method.key,
+		} );
+	}
+
 	render() {
 		const currentMethod = this.getCurrentMethod();
-		const { methods, query } = this.props;
-		const { enabledMethods, recommendedMethod } = this.state;
+		const { methods } = this.props;
+		const { enabledMethods, recommendedMethod, configuringMethods } = this.state;
 		const configuredMethods = methods.filter(
 			( method ) => method.isConfigured
 		).length;
 
-		if ( currentMethod ) {
-			return (
-				<Card className="woocommerce-task-payment-method is-narrow">
-					{ cloneElement( currentMethod.container, {
-						query,
-						installStep: this.getInstallStep(),
-						markConfigured: this.markConfigured,
-						hasCbdIndustry: currentMethod.hasCbdIndustry,
-					} ) }
-				</Card>
-			);
-		}
+		const setupElements = methods.map( ( method ) => {
```

Comment:
Those "setup elements" are supposed to be invisible if they're inline. For example, the WCPay setup element is completely invisible, so we don't need to redirect to a different "page" just to show the installation progress. If the user clicks in "Stripe", then the `setupElement` for Stripe will be displayed (along with any other invisible setup elements that were currently in progress) and nothing else.

It's a bit convoluted, but that's the best I could come up with while keeping the "declarative" list of methods configuration. I'd be happy to improve the naming of things if you can think of something less confusing :)

Reply:
I realize that the WCPay one is invisible, but figured a) conceptually it seemed more organized to keep the single list of methods and "inline" the setup as a child, and b) hypothetically there can be inline setup elements for other methods that aren't invisible – nothing in the implementation is hiding them and a UI is what's generally rendered by the method components.

If it's kept as it is, then I think there should be a comment somewhere either discouraging `hasInlineSetup` for actual UI or messaging, or else pointing out that it will be at the _top_ of the screen.

Reply:
Those are some good points. What do you think about making explicit that those elements will be invisible? https://github.com/woocommerce/woocommerce-admin/pull/4499/commits/0af201a22d44dd801e199264e2f1b44f9db88cdc

Reply:
That seems fine to me, though I'm starting to think it should be called "hidden setup", "auto setup", or "external setup", rather than "inline setup" – thoughts?

Reply:
"Auto setup" sounds good, thanks for giving several options 😂 

Changed in https://github.com/woocommerce/woocommerce-admin/pull/4499/commits/3f5f8af47120f76e2faec6c8dacefc8c91f24e8f

Inline setup made sense because you're setting it up without leaving the methods list page, but you're right, "inline" has a different meaning.